hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -970,6 +972,53 @@ ContainerCommandResponseProto handleWriteChunk(
     return getWriteChunkResponseSuccess(request, blockDataProto);
   }
 
+  /**
+   * Handle Write Chunk operation for closed container. Calls ChunkManager to process the request.
+   *
+   */
+  private void writeChunkForClosedContainer(ChunkInfo chunkInfo, BlockID blockID,
+                                            ChunkBuffer data, KeyValueContainer kvContainer,
+                                            BlockData blockData) {
+
+    try {
+      Preconditions.checkNotNull(chunkInfo);
+      Preconditions.checkNotNull(data);
+      long writeChunkStartTime = Time.monotonicNowNanos();
+      checkContainerClose(kvContainer);
+
+      DispatcherContext dispatcherContext = DispatcherContext.getHandleWriteChunk();
+
+      // Set CHUNK_OVERWRITE to overwrite existing chunk.
+      chunkInfo.addMetadata(OzoneConsts.CHUNK_OVERWRITE, "true");
+      chunkManager.writeChunk(kvContainer, blockID, chunkInfo, data,
+          dispatcherContext);
+
+      // Increment write stats for WriteChunk after write.
+      metrics.incContainerBytesStats(Type.WriteChunk, chunkInfo.getLen());
+      metrics.incContainerOpsLatencies(Type.WriteChunk, Time.monotonicNowNanos() - writeChunkStartTime);
+
+      if (blockData != null) {
+        long putBlockStartTime = Time.monotonicNowNanos();
+        // Add ChunkInfo to BlockData to be persisted in RocksDb
+        List<ContainerProtos.ChunkInfo> chunks = blockData.getChunks();
+        chunks.add(chunkInfo.getProtoBufMessage());
+        blockData.setChunks(chunks);
+
+        // To be set from the Replica's BCSId
+        blockData.setBlockCommitSequenceId(dispatcherContext.getLogIndex());
+
+        blockManager.putBlock(kvContainer, blockData);
```

Comment:
The `endOfBlock` boolean is implicitly set to true here. I think that's a quirk on the write path we don't need to worry about in this case and can explicitly set it to false.